airbyte-integrations/connectors/source-linnworks/source_linnworks/streams.py

```diff
@@ -0,0 +1,227 @@
+    ) -> str:
+        return "/api/Locations/GetLocation"
+
+    def request_params(
+        self, stream_state: Mapping[str, Any], stream_slice: Mapping[str, any] = None, next_page_token: Mapping[str, Any] = None
+    ) -> MutableMapping[str, Any]:
+        return {"pkStockLocationId ": stream_state["pkStockLocationId"]}
+
+
+class StockLocations(LinnworksStream):
+    # https://apps.linnworks.net/Api/Method/Inventory-GetStockLocations
+    # Response: List<StockLocation> https://apps.linnworks.net/Api/Class/linnworks-spa-commondata-Inventory-ClassBase-StockLocation
+    # Allows 150 calls per minute
+    primary_key = "StockLocationIntId"
+
+    def path(
+        self, stream_state: Mapping[str, Any] = None, stream_slice: Mapping[str, Any] = None, next_page_token: Mapping[str, Any] = None
+    ) -> str:
+        return "/api/Inventory/GetStockLocations"
+
+    def read_records(
+        self,
+        sync_mode: SyncMode,
+        cursor_field: List[str] = None,
+        stream_slice: Mapping[str, Any] = None,
+        stream_state: Mapping[str, Any] = None,
+    ) -> Iterable[Mapping[str, Any]]:
+        records = super().read_records(sync_mode, cursor_field, stream_slice, stream_state)
+
+        for record in records:
+            location = Location(authenticator=self.authenticator)
+            stock_location_records = location.read_records(
+                sync_mode, cursor_field, stream_slice, {"pkStockLocationId": record["StockLocationId"]}
+            )
+            record["location"] = next(stock_location_records)
+            yield record
+
+
+class StockItems(LinnworksStream):
+    # https://apps.linnworks.net//Api/Method/Stock-GetStockItemsFull
+    # Response: List<StockItemFull> https://apps.linnworks.net/Api/Class/linnworks-spa-commondata-Inventory-ClassBase-StockItemFull
+    # Allows 250 calls per minute
+    primary_key = "StockItemIntId"
+    page_size = 200
+
+    raise_on_http_errors = False
+
+    def path(
+        self, stream_state: Mapping[str, Any] = None, stream_slice: Mapping[str, Any] = None, next_page_token: Mapping[str, Any] = None
+    ) -> str:
+        return "/api/Stock/GetStockItemsFull"
+
+    def next_page_token(self, response: requests.Response) -> Optional[Mapping[str, Any]]:
+        url = urlparse(response.request.url)
+        qs = dict(parse_qsl(url.query))
+
+        page_size = int(qs.get("entriesPerPage", self.page_size))
+        page_number = int(qs.get("pageNumber", 0))
+
+        data = response.json()
+
+        if response.status_code == requests.codes.ok and len(data) == page_size:
+            return {
+                "entriesPerPage": page_size,
+                "pageNumber": page_number + 1,
+            }
+
+    def parse_response(self, response: requests.Response, **kwargs) -> Iterable[Mapping]:
+        if response.status_code != requests.codes.ok:
```

Comment:
When you request an empty page, is the HTTP status code specific (e.g 404)? It could be easier to isolate this “handled” error from others that might occur and remain silent.